packages/pnpm-sync-lib/src/pnpmSyncPrepare.ts

```diff
@@ -260,47 +267,25 @@ export async function pnpmSyncPrepareAsync(options: IPnpmSyncPrepareOptions): Pr
   });
 }
 
-// process dependencies and devDependencies to generate injectedDependencyToFilePath
-function getInjectedDependencyToVersion(pnpmLockfile: ILockfile | undefined): Map<string, Set<string>> {
-  const injectedDependencyToVersion: Map<string, Set<string>> = new Map();
-  for (const importerKey in pnpmLockfile?.importers) {
-    if (!pnpmLockfile?.importers[importerKey]?.dependenciesMeta) {
-      continue;
-    }
-    const dependenciesMeta = pnpmLockfile?.importers[importerKey]?.dependenciesMeta;
-
-    for (const dependency in dependenciesMeta) {
-      if (dependenciesMeta[dependency]?.injected) {
-        if (!injectedDependencyToVersion.has(dependency)) {
-          injectedDependencyToVersion.set(dependency, new Set());
-        }
-      }
-    }
-
-    // based on https://pnpm.io/package_json#dependenciesmeta
-    // the injected dependencies could available inside dependencies, optionalDependencies, and devDependencies.
-    processDependencies(pnpmLockfile?.importers[importerKey]?.dependencies, injectedDependencyToVersion);
-    processDependencies(pnpmLockfile?.importers[importerKey]?.devDependencies, injectedDependencyToVersion);
-    processDependencies(
-      pnpmLockfile?.importers[importerKey]?.optionalDependencies,
-      injectedDependencyToVersion
-    );
-  }
-  return injectedDependencyToVersion;
-}
-function processDependencies(
+function getInjectedDependencyToVersion(
   dependencies: Record<string, IVersionSpecifier> | undefined,
   injectedDependencyToVersion: Map<string, Set<string>>
 ): void {
   if (dependencies) {
     for (const [dependency, specifier] of Object.entries(dependencies)) {
-      if (injectedDependencyToVersion.has(dependency)) {
-        const specifierToUse: string = typeof specifier === 'string' ? specifier : specifier.version;
-
-        // the injected dependency should always start with file protocol
-        if (specifierToUse.startsWith('file:')) {
-          injectedDependencyToVersion.get(dependency)?.add(specifierToUse);
+      const specifierToUse: string = typeof specifier === 'string' ? specifier : specifier.version;
+      // the injected dependency should always start with file protocol
+      // and exclude tarball installation
+
+      const tarballSuffix = ['.tar', '.tar.gz', '.tgz'];
```

Comment:
How do we confirm that this is correct for the latest PNPM version?  Maybe add a `//` comment mentioning this API:

[@pnpm/lockfile-utils/src/refIsLocalTarball.ts](https://github.com/pnpm/pnpm/blob/f0b51bb444e5edfc5eec61dcc10e6775f9862f1c/lockfile/lockfile-utils/src/refIsLocalTarball.ts)
```ts
export function refIsLocalTarball (ref: string): boolean {
  return ref.startsWith('file:') && (ref.endsWith('.tgz') || ref.endsWith('.tar.gz') || ref.endsWith('.tar'))
}
```